### ISSUE TITLE
Support FOSSology on SUSE

### ISSUE THREAD

**Issue by yaobinshi** [fossology/fossology#308]
---

Author Name: **Dong Ma** 
Original Redmine Issue: 7095, http://www.fossology.org/issues/7095
Original Date: 2014/06/16
Original Assignee: Dong Ma

---

We have one user try our source code on SUSE, and done some notes here: http://www.fossology.org/projects/fossology/wiki/Installsuse

Also got some issues about pkgagent and upload from file, this backlog will trace the issues on SUSE.

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **Dong Ma** 
Original Date: 2014/06/16 07:29:02 +0000

---

pkgaegnt issue:

pkgagent.c: In function ‘ProcessUpload’:
pkgagent.c:343: warning: implicit declaration of function ‘rpmFreeCrypto’
pkgagent.c: In function ‘ReadHeaderInfo’:
pkgagent.c:374: error: storage size of ‘req’ isn’t known
pkgagent.c:375: error: ‘rpm_count_t’ undeclared (first use in this function)
pkgagent.c:375: error: (Each undeclared identifier is reported only once
pkgagent.c:375: error: for each function it appears in.)
pkgagent.c:375: error: expected ‘;’ before ‘data_size’
pkgagent.c:450: warning: implicit declaration of function ‘headerGet’
pkgagent.c:450: error: ‘HEADERGET_DEFAULT’ undeclared (first use in this function)
pkgagent.c:452: error: ‘data_size’ undeclared (first use in this function)
pkgagent.c:452: warning: implicit declaration of function ‘rpmtdCount’
pkgagent.c:455: warning: implicit declaration of function ‘rpmtdNextString’
pkgagent.c:455: warning: initialization makes pointer from integer without a cast
pkgagent.c:460: warning: implicit declaration of function ‘rpmtdFreeData’
pkgagent.c:374: warning: unused variable ‘req’
make[3]: *** [pkgagent.o] Error 1
pkgagent.c: In function ‘ProcessUpload’:
pkgagent.c:343: warning: implicit declaration of function ‘rpmFreeCrypto’
pkgagent.c: In function ‘ReadHeaderInfo’:
pkgagent.c:374: error: storage size of ‘req’ isn’t known
pkgagent.c:375: error: ‘rpm_count_t’ undeclared (first use in this function)
pkgagent.c:375: error: (Each undeclared identifier is reported only once
pkgagent.c:375: error: for each function it appears in.)
pkgagent.c:375: error: expected ‘;’ before ‘data_size’
pkgagent.c:450: warning: implicit declaration of function ‘headerGet’
pkgagent.c:450: error: ‘HEADERGET_DEFAULT’ undeclared (first use in this function)
pkgagent.c:452: error: ‘data_size’ undeclared (first use in this function)
pkgagent.c:452: warning: implicit declaration of function ‘rpmtdCount’
pkgagent.c:455: warning: implicit declaration of function ‘rpmtdNextString’
pkgagent.c:455: warning: initialization makes pointer from integer without a cast
pkgagent.c:460: warning: implicit declaration of function ‘rpmtdFreeData’
pkgagent.c:374: warning: unused variable ‘req’
make[3]: *** [pkgagent.o] Error 1
pkgagent.c: In function ‘ProcessUpload’:
pkgagent.c:343: warning: implicit declaration of function ‘rpmFreeCrypto’
pkgagent.c: In function ‘ReadHeaderInfo’:
pkgagent.c:374: error: storage size of ‘req’ isn’t known
pkgagent.c:375: error: ‘rpm_count_t’ undeclared (first use in this function)
pkgagent.c:375: error: (Each undeclared identifier is reported only once
pkgagent.c:375: error: for each function it appears in.)
pkgagent.c:375: error: expected ‘;’ before ‘data_size’
pkgagent.c:450: warning: implicit declaration of function ‘headerGet’
pkgagent.c:450: error: ‘HEADERGET_DEFAULT’ undeclared (first use in this function)
pkgagent.c:452: error: ‘data_size’ undeclared (first use in this function)
pkgagent.c:452: warning: implicit declaration of function ‘rpmtdCount’
pkgagent.c:455: warning: implicit declaration of function ‘rpmtdNextString’
pkgagent.c:455: warning: initialization makes pointer from integer without a cast
pkgagent.c:460: warning: implicit declaration of function ‘rpmtdFreeData’
pkgagent.c:374: warning: unused variable ‘req’
make[3]: *** [pkgagent.o] Error 1
pkgagent.c: In function ‘ProcessUpload’:
pkgagent.c:343: warning: implicit declaration of function ‘rpmFreeCrypto’
pkgagent.c: In function ‘ReadHeaderInfo’:
pkgagent.c:374: error: storage size of ‘req’ isn’t known
pkgagent.c:375: error: ‘rpm_count_t’ undeclared (first use in this function)
pkgagent.c:375: error: (Each undeclared identifier is reported only once
pkgagent.c:375: error: for each function it appears in.)
pkgagent.c:375: error: expected ‘;’ before ‘data_size’
pkgagent.c:450: warning: implicit declaration of function ‘headerGet’
pkgagent.c:450: error: ‘HEADERGET_DEFAULT’ undeclared (first use in this function)
pkgagent.c:452: error: ‘data_size’ undeclared (first use in this function)
pkgagent.c:452: warning: implicit declaration of function ‘rpmtdCount’
pkgagent.c:455: warning: implicit declaration of function ‘rpmtdNextString’
pkgagent.c:455: warning: initialization makes pointer from integer without a cast
pkgagent.c:460: warning: implicit declaration of function ‘rpmtdFreeData’
pkgagent.c:374: warning: unused variable ‘req’
make[3]: *** [pkgagent.o] Error 1

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **Dong Ma** 
Original Date: 2014/06/16 07:31:29 +0000

---

Mark Felton  Note about 2.5 tar file:

SUSE 11 Fossology 2.5 tar file (can not use RPM)

1. Download fossology 2.5 tar and untar

2. Using Yast
   apache2
   postregsql
   php
   magic

  Be sure to include development packages
  Be sure to include postgresql server

3. In data fix pg_hba.conf
  Add local all all 
  File is in 
  /var/lib/pgsql/data 

4 Download a correct version of magic.h
    Put in appropriate directory and fix the c files in fossology to point to it

5. Fix function __sync_fetch_and_and in scheduler.c
   To fix this I did an undef of __GNUC__
   This made the compile go through ok

6. In pkagent.c removed the _RPM_4_4_COMPAT code

7. Create symbolic link to lib magic (not sure if there is a better way to do this)

8. Had to fix some path issues (again not sure if there is another way)
    I did some with symbolic links and some by changing permissions

9. Had to handle some space issues. I linked /srv/fossology to an area with adequate space.
